mmv1/third_party/terraform/website/docs/r/compute_instance_template.html.markdown

```diff
@@ -271,6 +271,15 @@ The following arguments are supported:
 
 * `confidential_instance_config` (Optional) - Enable [Confidential Mode](https://cloud.google.com/compute/confidential-vm/docs/about-cvm) on this VM.
 
+* `network_performance_config` (Optional) Configures [network performance settings](https://cloud.google.com/compute/docs/networking/configure-vm-with-high-bandwidth-configuration)
+  for the instance. Structure is documented below. **Note**: [`machine_type`](#machine_type)
+  must be a [supported type](https://cloud.google.com/compute/docs/networking/configure-vm-with-high-bandwidth-configuration#bandwidth_tiers),
+  the [`image`](#image) used must include [`GVNIC`](https://cloud.google.com/compute/docs/networking/using-gvnic#create-instance-gvnic-image)
+  in `guest-os-features`, and `network_interface.0.nic-type` must be `GVNIC`
+  in order for this setting to take effect.
```

Comment:
Given the tight set of constraints for when this argument actually can take effect, I wasn't sure if we want to build any validation functions to fail fast if those conditions aren't met or if we're best to leave that to the API. I hedge on thinking the API should be responsible for those error states or input combinations that don't make sense.

Reply:
I would agree. In addition, we've had similar validation functions become blockers in situations where the API has changed its own requirements.

Reply:
Thanks for the validation on my validation thought 😆